taker/src/lib.rs

```
@@ -0,0 +1,539 @@
+    /// This **only** has an effect when `json` is also enabled.
+    #[clap(long)]
+    pub json_span_list: bool,
+
+    /// If enabled, traces will be exported to the OTEL collector
+    #[clap(long)]
+    instrumentation: bool,
+
+    /// If enabled, tokio runtime can be locally debugged with tokio_console
+    #[clap(long)]
+    pub tokio_console: bool,
+
+    /// If enabled, libp2p ping and offer broadcast spans will be included in the traces exported
+    /// by the application.
+    #[clap(long)]
+    pub verbose_spans: bool,
+
+    /// OTEL collector endpoint address
+    ///
+    /// If not specified it defaults to the local collector endpoint.
+    #[clap(long, default_value = LOCAL_COLLECTOR_ENDPOINT )]
+    collector_endpoint: String,
+
+    /// If enabled, browser UI is not automatically launched at startup.
+    #[clap(long)]
+    pub headless: bool,
+
+    /// Service name for OTEL.
+    ///
+    /// If not specified it defaults to the binary name.
+    #[clap(long, default_value = "taker")]
+    service_name: String,
+
+    /// Configure the log level, e.g.: one of Error, Warn, Info, Debug, Trace
+    #[clap(short, long, default_value = "Debug")]
+    log_level: LevelFilter,
+
+    /// Password for the web interface.
+    ///
+    /// If not provided, will be derived from the seed.
+    #[clap(long)]
+    password: Option<Password>,
+
+    #[clap(subcommand)]
+    network: Option<Network>,
+
+    #[clap(short, long, parse(try_from_str = parse_app_seed))]
+    app_seed: Option<[u8; 32]>,
+
+    /// If provided will be used for internal wallet instead of a random key or app_seed. The
+    /// keys will be derived according to Bip84.
+    #[clap(short, long)]
+    pub wallet_xprv: Option<ExtendedPrivKey>,
+}
+
+impl Opts {
+    pub fn read() -> Opts {
+        Opts::parse()
+    }
+
+    pub fn new(network: String, password: String, data_dir: String) -> Opts {
```

🔧 I think it would be good to parse the `network` and `password` at the top and convert them into the types we want or fail early.